Code/GraphMol/MolOps.cpp

```diff
@@ -264,8 +302,19 @@ void cleanUp(RWMol &mol) {
 }
 
 void cleanUpOrganometallics(RWMol &mol) {
-  for (auto atom : mol.atoms()) {
-    metalBondCleanup(mol, atom);
+  std::vector<unsigned int> ranks(mol.getNumAtoms());
+  RDKit::Canon::rankMolAtoms(mol, ranks);
```

Comment:
Given that `rankMolAtoms` is an expensive operation, I think it makes sense to ensure that there actually is a metal-hypervalent_atom bond before proceeding in this function.